pkg/watcherapplier/statefulset.go

```diff
@@ -0,0 +1,137 @@
+package watcherapplier
+
+import (
+	"github.com/openstack-k8s-operators/lib-common/modules/common/env"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	watcherv1beta1 "github.com/openstack-k8s-operators/watcher-operator/api/v1beta1"
+	watcher "github.com/openstack-k8s-operators/watcher-operator/pkg/watcher"
+)
+
+const (
+	// ServiceCommand -
+	ServiceCommand = "/usr/local/bin/kolla_start"
+
+	// ComponentName -
+	ComponentName = watcher.ServiceName + "-applier"
+)
+
+// StatefulSet - returns the StatefulSet definition for the watcher-applier service
+func StatefulSet(
+	instance *watcherv1beta1.WatcherApplier,
+	configHash string,
+	labels map[string]string,
+	annotations map[string]string,
+) *appsv1.StatefulSet {
+	runAsUser := int64(0)
+	var config0644AccessMode int32 = 0644
+
+	envVars := map[string]env.Setter{}
+	envVars["KOLLA_CONFIG_STRATEGY"] = env.SetValue("COPY_ALWAYS")
+	envVars["CONFIG_HASH"] = env.SetValue(configHash)
+	args := []string{"-c", ServiceCommand}
+
+	livenessProbe := &corev1.Probe{
+		// TODO might need tuning
+		TimeoutSeconds: 10,
+		PeriodSeconds:  10,
+	}
+	readinessProbe := &corev1.Probe{
+		// TODO might need tuning
+		TimeoutSeconds: 5,
+		PeriodSeconds:  5,
+	}
+
+	livenessProbe.Exec = &corev1.ExecAction{
+		Command: []string{
+			"/usr/bin/pgrep", "-r", "DRST", "watcher-applier",
```

Comment:
is this a good approach for livenessprobe with watcher-applier?

Reply:
yep we just make sure the process is not dead.

until there are native health endpoint in each process we should not do anything else more advanced.

the only exception too this is the API where we can actually use a HTTP probe to hit the microverion endpoint 

//
	livenessProbe.HTTPGet = &corev1.HTTPGetAction{
		Port: intstr.IntOrString{Type: intstr.Int, IntVal: int32(PlacementPublicPort)},
	}

Reply:
since that is OK (adding same in decisionengine) you could use the ComponentName constant instead of hard-coding "watcher-applier" here